### ISSUE TITLE
Cannot find 'ou' method

### ISSUE THREAD

**Issue by dfduarte** [TAMUArch/cookbook.windows_ad#56]
Hello,

Recently, I did a little wrapper to insert a machine in my AD, and to do some tests with this cookbook. Everything ok...

But, when I add the ou attribute to join the server within a OU, I receive the following message:

```
Relevant File Content:
----------------------
c:/chef/cache/cookbooks/ac-windows-setup/recipes/default.rb:

121:  end
122:
123:
124:  windows_ad_domain "vmac.com" do
125:    action :join
126:    domain_pass "vagrant"
127:    domain_user "vagrant"
128>>   ou "test01/test"
129:  end
130:
131:  =begin
132:  also, this cookbook uses windows_ad, in order to control the active direct
ory functions.
133:
134:  https://supermarket.chef.io/cookbooks/windows_ad
135:
136:  =end
137:


Running handlers:
[2015-10-16T14:20:39-07:00] ERROR: Running exception handlers
Running handlers complete
[2015-10-16T14:20:39-07:00] ERROR: Exception handlers complete
Chef Client failed. 0 resources updated in 13 seconds
[2015-10-16T14:20:39-07:00] FATAL: Stacktrace dumped to c:/chef/cache/chef-stack
trace.out
[2015-10-16T14:20:39-07:00] FATAL: NoMethodError: undefined method `ou' for LWRP
 resource windows_ad_domain from cookbook windows_ad

C:\Users\vagrant>
```

And I can't join the machine in AD alongside the OU.

I see that the logic is already implemented in the domain.rb resource, but it's not working properly.

**Comment by derekgroh** [comment]
@dfduarte Can you provide your recipe and I'd be happy to help you troubleshoot?  Also please include what version of the cookbook you are using.

**Comment by dfduarte** [comment]
Hello @derekgroh 

So far, my recipe is just a powershell script that changes the DNS configuration from all the network interfaces, and points the primary to AD, and joins the machine to Directory:

```
powershell_script 'Change_bulk_dns_fix_ad_lookup' do
  code <<-EOH

        $NICs = Get-WMIObject Win32_NetworkAdapterConfiguration | Where {$_.IPEnabled -eq "TRUE"}
        
          Foreach($NIC in $NICs) {
            $DNSServers = "192.168.0.116","8.8.8.8"
            $NIC.SetDNSServerSearchOrder($DNSServers)
            $NIC.SetDynamicDNSRegistration("TRUE")
            $NIC.SetWINSServer("", "")
        
          }

          Start-Sleep -s 10

          EOH
end


windows_ad_domain "vmac.com" do
  action :join
  domain_pass "vagrant"
  domain_user "vagrant"
  ou "test01"
end
```
(I've deleted the test01/test, and keept just the test01. It was just a test to see this error. The correct OU is test01)

**Comment by dfduarte** [comment]
Oh, sorry: The version is 0.4.2!

**Comment by derekgroh** [comment]
What version of chef-client are you using?  I was able to join successfully on 11.18.12

**Comment by dfduarte** [comment]
@derekgroh I'm using the lastest (12.5.2) here. I simply bootstrapped some new machines and started to test the cookbook.

**Comment by dfduarte** [comment]
Hello @derekgroh 

Btw: I tired to run the cookbook with your 11.18.12 version. but my cookbook uses the "Poise" library. I use it to write some providers for my cookbooks. Poise is not compatible with Chef < 12 (as up I can see), it throws a lot of errors in libraries and recipes.

So, I tried the chef 12.3, given that 12.4 implemented A LOT of changes in DSLs and providers processing. But, I had the same error =\.

**Comment by dfduarte** [comment]
Hey @derekgroh,

I guess that I figured out what is the problem:

For some reason, the version that I have on my Chef-Server, don't have declared the resource "ou" on resource/domain.rb. Maybe, the problem can be solved declaring this.

Strange fact: my git has cloned your repository as 0.4.3v. And this problem were fixed on this version. Did you fixed that in the meantime?

**Comment by derekgroh** [comment]
@dfduarte ok had to unshare the cookbook from supermarket to get it to correct the issue.  Will attempt to figure out how I can upload past versions over the next few days.  See if the latest version from supermaket works for you

**Comment by dfduarte** [comment]
Hello @derekgroh 

I did some tests here, and the new 0.4.4 version worked like a Swiss clock. Great work my friend =)! (At least, on the parts which I need on your library/cookbook. But everything seems working ok)

I just found a little trick problem. But, it's more a problem from Win2008R2 NETDOM than the cookbook itself: The command fails if you use a password with semi-collon. Yep, this is not a issue from cookbook, just a tip.

Anyway, thanks dude =)
